libs/sdk-ui-dashboard/src/presentation/drill/DrillConfigPanel/DrillToUrl/CustomUrlEditor.tsx

```diff
@@ -0,0 +1,266 @@
+}
+
+const initialCursorPosition: ICursorPosition = {
+    from: 0,
+    to: 0,
+};
+
+const insertPlaceholderAtCursor = (text: string, placeholder: string, cursor: ICursorPosition): string =>
+    `${text.substring(0, cursor.from)}${placeholder}${text.substring(cursor.to)}`;
+
+const assertValidUrl = (url: string) =>
+    /^[A-Za-z0-9.\-+]+:|^\{attribute_title\(/.test(url) ? url : `https://${url}`;
+
+const getWarningTextForInvalidParameters = (parameters: string[]): React.ReactElement => {
+    const invalidParameters = parameters.map((parameter) => `"${parameter}"`).join(", ");
+    return (
+        <FormattedMessage
+            id="configurationPanel.drillIntoUrl.editor.invalidAttributeDisplayForms"
+            values={{ invalidParameters }}
+        />
+    );
+};
+
+export interface CustomUrlEditorProps {
+    urlDrillTarget?: UrlDrillTarget;
+    attributeDisplayForms?: IAttributeDisplayFormMetadataObject[];
+    loadingAttributeDisplayForms: boolean;
+    invalidAttributeDisplayFormIdentifiers: string[];
+    documentationLink?: string;
+    enableClientIdParameter: boolean;
+    enableDataProductIdParameter: boolean;
+    enableWidgetIdParameter: boolean;
+    onClose: () => void;
+    onSelect: (customUrl: string) => void;
+}
+
+const CustomUrlEditorDialog: React.FunctionComponent<CustomUrlEditorProps> = (props) => {
+    const {
+        urlDrillTarget,
+        documentationLink,
+        onSelect,
+        onClose,
+        attributeDisplayForms,
+        loadingAttributeDisplayForms,
+        invalidAttributeDisplayFormIdentifiers,
+        enableClientIdParameter,
+        enableDataProductIdParameter,
+        enableWidgetIdParameter,
+    } = props;
+
+    const intl = useIntl();
+
+    const previousValue = urlDrillTarget
+        ? (isDrillToCustomUrlConfig(urlDrillTarget) && urlDrillTarget.customUrl) || ""
+        : "";
+
+    const [currentValue, setCurrentValue] = useState(previousValue);
+    const apply = () => onSelect(assertValidUrl(currentValue));
+    const handleOnChange = (value: string) => setCurrentValue(value.trim());
+
+    const isApplyEnabled = currentValue && currentValue.localeCompare(previousValue) !== 0;
+
+    const [cursorPosition, setCursorPosition] = useState<ICursorPosition>(initialCursorPosition);
+    const handleCursorPosition = (from: number, to: number) => setCursorPosition({ from, to });
+
+    const handleOnAdd = (parameterPlaceholder: string) =>
+        setCurrentValue(insertPlaceholderAtCursor(currentValue, parameterPlaceholder, cursorPosition));
+
+    const editorWarningText =
+        invalidAttributeDisplayFormIdentifiers.length > 0
+            ? getWarningTextForInvalidParameters(invalidAttributeDisplayFormIdentifiers)
+            : undefined;
+
+    return (
+        <ConfirmDialogBase
+            className="gd-drill-custom-url-editor s-gd-drill-custom-url-editor"
+            isPositive={true}
+            headline={intl.formatMessage({
+                id: previousValue
```

Comment:
This will not be picked up by the i18n validator, please put the condition outside of the `intl.formatMessage` call.